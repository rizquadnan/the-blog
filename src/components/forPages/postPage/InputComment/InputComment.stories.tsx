import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { InputComment } from './InputComment'
import { action } from '@storybook/addon-actions'

type TInputComment = React.ComponentProps<typeof InputComment>
export default {
  title: 'User Inputs/InputComment',
  component: InputComment,
} as Meta<TInputComment>

const Template: Story<TInputComment> = (args) => <InputComment {...args} />

export const Default = Template.bind({})
Default.args = {
  onSubmit: action('Submit!'),
}
