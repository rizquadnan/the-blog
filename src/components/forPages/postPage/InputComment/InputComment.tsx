import { SendIcon } from '@/components/uiKit/Icons'
import { Box, IconButton, Textarea } from '@chakra-ui/react'
import React, { useState } from 'react'

type TInputComment = {
  onSubmit(value: string): void
  isDisabled?: boolean
}

export function InputComment(props: TInputComment) {
  const [value, setValue] = useState('')

  return (
    <Box position="relative">
      <Textarea
        value={value}
        placeholder="Add comment"
        onChange={(e) => setValue(e.target.value)}
        paddingRight="60px !important"
        isDisabled={props.isDisabled}
      />
      <IconButton
        position="absolute"
        top="50%"
        right="12px"
        transform="translateY(-50%)"
        opacity={value ? 1 : 0}
        transition="opacity 250ms"
        onClick={() => props.onSubmit(value)}
        variant="ghost"
        aria-label="Submit"
        colorScheme="orange"
        icon={<SendIcon />}
        zIndex={2}
      />
    </Box>
  )
}

export default InputComment
