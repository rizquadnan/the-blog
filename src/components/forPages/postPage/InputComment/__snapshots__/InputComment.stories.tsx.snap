// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Inputs/InputComment Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "multiline": true,
      "name": "Add comment",
      "pressed": undefined,
      "role": "textbox",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Submit",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "user-inputs-inputcomment--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
